packages/core/src/DatadogProvider/__tests__/initialization.test.tsx

```diff
@@ -0,0 +1,99 @@
+
+jest.mock('../../TimeProvider', () => {
+    const now = jest.fn();
+    return {
+        TimeProvider: jest.fn().mockImplementation(() => {
+            return { now };
+        })
+    };
+});
+const nowMock = new TimeProvider().now;
+
+const flushPromises = () => new Promise<void>(setImmediate);
+
+describe('DatadogProvider', () => {
+    afterEach(() => {
+        jest.clearAllMocks();
+        DdSdkReactNative['wasInitialized'] = false;
+        DatadogProvider.isInitialized = false;
+        BufferSingleton.reset();
+        (nowMock as any).mockReturnValue('timestamp_not_specified');
+    });
+    describe('initialization', () => {
+        it('renders its children and initializes the SDK once', async () => {
+            const {
+                getByText,
+                rerenderWithRandomConfig
+            } = renderWithProvider();
+            getByText('I am a test application');
+            expect(NativeModules.DdSdk.initialize).toHaveBeenCalledTimes(1);
+
+            // We remove the sdk version from the configuration as it would require to update this snapshot
+            const receivedConfiguration =
+                NativeModules.DdSdk.initialize.mock.calls[0][0];
+            delete receivedConfiguration.additionalConfig['_dd.sdk_version'];
+            expect(receivedConfiguration).toMatchInlineSnapshot(`
+                  DdSdkConfiguration {
+                    "additionalConfig": Object {
+                      "_dd.first_party_hosts": Array [],
+                      "_dd.long_task.threshold": 200,
+                      "_dd.native_view_tracking": false,
+                      "_dd.source": "react-native",
+                    },
+                    "applicationId": "fakeApplicationId",
+                    "clientToken": "fakeToken",
+                    "env": "fakeEnv",
+                    "nativeCrashReportEnabled": false,
+                    "sampleRate": 100,
+                    "site": "US",
+                    "trackingConsent": "granted",
+                  }
+            `);
+
+            // Re-render
+            rerenderWithRandomConfig();
+            expect(NativeModules.DdSdk.initialize).toHaveBeenCalledTimes(1);
+        });
+
+        it('keeps events in the buffer then executes the buffer once initialization is done', async () => {
+            // Given
+            NativeModules.DdTrace.startSpan.mockReturnValueOnce('good_span_id');
+            (nowMock as any).mockReturnValue('good_timestamp');
+            await DdRum.addAction(RumActionType.TAP, 'fakeAction');
+
+            // When
+            const spanId = await DdTrace.startSpan('fakeOperation');
+            await DdTrace.finishSpan(spanId);
+            (nowMock as any).mockReturnValue('bad_timestamp');
+
+            renderWithProvider();
+            await flushPromises();
+
+            // Then
+            expect(NativeModules.DdSdk.initialize).toHaveBeenCalledTimes(1);
```

Comment:
Does this test that the events are held in the buffer until completion, or just that they were called? Is there any check to make sure calls haven't happened prior to initialization?

Reply:
good call, I'll improve this test 👍